dsa/src/components.rs

```diff
@@ -63,7 +68,7 @@ impl Components {
 
     /// Check whether the components are valid
     #[must_use]
-    pub fn is_valid(&self) -> bool {
+    pub(crate) fn is_valid(&self) -> bool {
```

Comment:
Perhaps just factor this method into the constructor?

Reply:
Thanks for noticing this. I moved all the validation checks into the respective constructors